lib/datadog/tracing/contrib/sinatra/env.rb

```diff
@@ -40,24 +38,15 @@ def header_to_rack_header(name)
             "HTTP_#{name.to_s.upcase.gsub(/[-\s]/, '_')}"
           end
 
-          # Was a Sinatra already traced in this request?
-          # We don't want to create spans for intermediate Sinatra
-          # middlewares that don't match the request at hand.
-          def middleware_traced?(env)
-            env[Ext::RACK_ENV_MIDDLEWARE_TRACED]
-          end
-
-          def set_middleware_traced(env, bool)
-            env[Ext::RACK_ENV_MIDDLEWARE_TRACED] = bool
-          end
+          def route_path(env, use_script_names: Datadog.configuration.tracing[:sinatra][:resource_script_names])
+            return unless env['sinatra.route']
 
-          # The start time of the top-most Sinatra middleware.
-          def middleware_start_time(env)
-            env[Ext::RACK_ENV_MIDDLEWARE_START_TIME]
-          end
-
-          def set_middleware_start_time(env, time = Time.now.utc)
-            env[Ext::RACK_ENV_MIDDLEWARE_START_TIME] = time
+            _, path = env['sinatra.route'].split(' ', 2)
+            if use_script_names
+              env[::Rack::SCRIPT_NAME].to_s + path
```

Comment:
I'm curious about this change -- we previously used `request.script_name` but now we directly parse it from the env. Could you add some context on why this option is needed?

Reply:
This does not change the behaviour of our instrumentation.

`Sinatra::Request` is a children of `Rack::Request`, while `#script_name` is still delegate to rack env with `SCRIPT_NAME`. I changed the implementation depends on what kind of object I could access within the current scope.

This option is introduce in [here](https://github.com/DataDog/dd-trace-rb/pull/283). 

Personally, I think this option should move to rack instrumentation and enabled(true) by default.

Reply:
This would be a breaking change(resource change), so probably plan for the next major release.

Reply:
Well, I wouldn't call it a breaking change. As the famous xkcd comic states:

![](https://imgs.xkcd.com/comics/workflow.png)

...every change has the potential to break someone. So if we applied it in the extreme every change is a potential breaking change. New integration? There's more spans now -- it's different. Fixed a typo in a resource? Someone relied on the typo -- it's different.

So I wouldn't consider that a breaking change, or at least one that would need waiting until the next major version.

But... this is just a suggestion, e.g. you're welcome to disagree and since tracing is your domain, you have more say in it than I do ;)